pkg/controller/datadogagent/utils.go

```diff
@@ -159,6 +159,24 @@ func getAgentContainer(dda *datadoghqv1alpha1.DatadogAgent) (*corev1.Container,
 	if err != nil {
 		return nil, err
 	}
+
+	udpPort := corev1.ContainerPort{
+		ContainerPort: datadoghqv1alpha1.DefaultDogstatsdPort,
+		Name:          "dogstatsdport",
+		Protocol:      corev1.ProtocolUDP,
+	}
+
+	if agentSpec.Config.HostPort != nil {
+		udpPort.HostPort = *agentSpec.Config.HostPort
+		if agentSpec.HostNetwork {
+			udpPort.ContainerPort = *agentSpec.Config.HostPort
+			envVars = append(envVars, corev1.EnvVar{
+				Name:  datadoghqv1alpha1.DDDogstatsdPort,
+				Value: strconv.Itoa(int(*agentSpec.Config.HostPort)),
+			})
+		}
+	}
```

Comment:
would be nice to have a comment explaining why we're doing this

Reply:
Agreed, will add it now! :)